components/feature/accounts/src/main/assets/extensions/fxawebchannel/background.js

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+Establish communication with native application.
+*/
+let port = browser.runtime.connectNative("mozacWebchannelBackground");
+/*
+Handle messages from native application, register content script for specific url.
+*/
+port.onMessage.addListener( event => {
+  if(event.type == "overrideFxAServer"){
+    browser.contentScripts.register({
+      "matches": [ event.url+"/*" ],
+      "js": [{file: "fxawebchannel.js"}],
+      "runAt": "document_start"
+    });
+    port.disconnect();
+  }
```

Comment:
So, just to make sure I understand: this is a one-shot communication channel that lets the application signal to the extension, that it is using a custom FxA server at the given url. Is that right?

Reply:
yes , you're right.